include/circt/Dialect/LTL/LTLOps.td

```diff
@@ -0,0 +1,312 @@
+      sequence matches if `%seq` matches 2, 3, or 4 cycles in the future.
+    - `ltl.delay %seq, 2` delays `%seq` by 2 or more cycles. The number of
+      cycles is unbounded but finite, which means that `%seq` *has* to match at
+      some point, instead of effectively never occuring by being delayed an
+      infinite number of cycles.
+    - `ltl.delay %seq, 0, 0` is equivalent to just `%seq`.
+
+    #### Clocking
+
+    The cycle delay specified on the operation refers to a clocking event. This
+    event is not directly specified by the delay operation itself. Instead, the
+    [`ltl.clock`](#ltlclock-circtltlclockop) operation can be used to associate
+    all delays within a sequence with a clock.
+  }];
+}
+
+def ConcatOp : LTLOp<"concat", [Pure]> {
+  let arguments = (ins Variadic<LTLAnySequenceType>:$inputs);
+  let results = (outs LTLSequenceType:$result);
+  let assemblyFormat = [{
+    $inputs attr-dict `:` type($inputs)
+  }];
+  let hasFolder = 1;
+  let hasCanonicalizeMethod = 1;
+
+  let summary = "Concatenate sequences into a longer sequence.";
+  let description = [{
+    Concatenates all of the `$inputs` sequences one after another into one
+    longer sequence. The sequences are arranged such that the end time of the
+    previous sequences coincides with the start time of the next sequence. This
+    means there is no implicit cycle of delay between the concatenated
+    sequences, which may be counterintuitive.
+
+    If a sequence should follow in the cycle after another sequence finishes,
+    that cycle of delay needs to be explicit. For example, *"u followed by v in
+    next cycle"* (`u ##1 v` in SVA) is represented as
+    `concat(u, delay(v, 1, 0))`:
+    ```
+    %0 = ltl.delay %v, 1, 0 : i1
+    ltl.concat %u, %v : !ltl.sequence, !ltl.sequence
+    ```
+    The resulting sequence checks for `u` in the first cycle and `v` in the
+    second, `[u, v]` in short.
+
+    Without this explicit delay, the previous sequence's end overlaps with the
+    next sequence's start. For example, consider the two sequences `u = a ##1 b`
+    and `v = c ##1 d`, which check for `a` and `c` in the first, and `b` and `d`
+    in the second cycle. When these two sequences are concatenated,
+    `concat(u, v)`, the end time of the first sequence coincides with the start
+    time of the second. As a result, the check for `b` at the end of the first
+    sequence will coincide with the check for `c` at the start of the second
+    sequence: `concat(u, v) = a ##1 (b && c) ##1 d`. The resulting sequence
+    checks for `a` in the first cycle, `b` and `c` in the second, and `d` in the
+    third, `[a, (b && c), d]` in short.
+
+    By making the delay between concatenated sequences explicit, the `concat`
+    operation behaves nicely in the presence of zero-length sequences. An empty,
+    zero-length sequence in a concatenation behaves as if the sequence wasn't
```

Comment:
This is a fair point. To be that jerk who brings up monoids, you indeed want there to be an identity element in your monoid.

Reply:
Something SV gets wrong almost everywhere 😉